### PR TITLE
Fix minimum, maximum forward-ad formula for float32

### LINKED ISSUE #69034: forward-mode AD: minimum, min, maximum, max formulas are loudly incorrect
## 🐛 Bug

Using forward-mode AD with `{minimum, maximum}` and fp32 Tensors raises an error

## To Reproduce

```
import torch
import torch.autograd.forward_ad as fwAD

x = torch.randn(3)
y = torch.randn(3)
tx = torch.randn(3)
ty= torch.randn(3)

with fwAD.dual_level():
  x_dual = fwAD.make_dual(x, tx)
  y_dual = fwAD.make_dual(y, ty)
  torch.maximum(x_dual, y_dual)
```
Gives: `RuntimeError: expected scalar type double but found float`

## Expected behavior

No error

## Environment

PyTorch main branch

cc @ezyang @albanD @zou3519 @gqchen @pearu @nikitaved @soulitzer @Lezcano @Varal7

### PR DESCRIPTION
**Pull request by zou3519**
fixes https://github.com/pytorch/pytorch/issues/69034

Test Plan:
- new tests

Fixes #ISSUE_NUMBER

### CODE REVIEW

File: tools/autograd/derivatives.yaml [hunk 1]
```diff
@@ -983,7 +983,7 @@
 - name: maximum(Tensor self, Tensor other) -> Tensor
   self: at::where(self == other, grad / 2, grad).masked_fill_(self < other, 0)
   other: at::where(self == other, grad / 2, grad).masked_fill_(self > other, 0)
-  result: other_t + at::where(self_p == other_p, 0.5, (self_p > other_p).to(result.scalar_type())) * (self_t - other_t)
```
Comment (soulitzer) [inline tools/autograd/derivatives.yaml:986]: Does this work? From `where`'s docs, it looks like it does not support float32 as the dtype for the tensor. Maybe we could use masked_fill instead?

> Currently valid scalar and tensor combination are 1. Scalar of floating dtype and torch.double 2. Scalar of integral dtype and torch.long 3. Scalar of complex dtype and torch.complex128

https://pytorch.org/docs/stable/generated/torch.where.html#torch.where

Comment (zou3519) [reply tools/autograd/derivatives.yaml:986]: Yes, the tests pass locally. Below is some visual evidence that torch.where supports tensors with float32 dtype.

image

The documentation is saying "IF you pass in a Python Number (Scalar) that is floating point, THEN any tensors need to be torch.double". Here's some evidence for that:
image

File: test/test_autograd.py [hunk 2]
```diff
@@ -7052,6 +7052,33 @@ def fn():
         # context manager. If you do, there is something wrong with the
         # locking of the forward ad level most likely
 
+    def test_maximum_minimum(self):
```
Comment (soulitzer) [inline test/test_autograd.py:7055]: We've been moving op-specific gradient tests out of test_autograd, maybe we could put this one in test_binary_ufuncs?

Comment (zou3519) [reply test/test_autograd.py]: I moved the test to test_binary_ufuncs, thanks for the suggestion!

**Review by soulitzer — APPROVED** [review]

**Pull request closed without merging**